File: contest/src/Utils/AbstractObservable.java
package Utils;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Sebi on 25-Nov-16.
 */
public abstract class AbstractObservable<E> implements Observable<E> {
    /*
    Observer list
    */
    private List<Observer> observers = new ArrayList<>();

    @Override
    public void addObserver(Observer<E> o){
        observers.add(o);
    }

    @Override
    public void removeObserver(Observer<E> o){
        observers.remove(o);
    }

    @Override
    public void notifyObservers(){
        for (Observer<E> o : observers){
            o.update();
        }
    }
}

File: contestC#/contest/contest/Utils/Observer/AbstractObservable.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace contest.Utils.Observer
{
    public class AbstractObservable : Observable
    {
        /*
         * Observer list 
         */
        List<Observer> observers = new List<Observer>();

        /*
         * Adds an observer
         */
        public void addObserver(Observer o)
        {
            observers.Add(o);
        }

        /*
         * Notifies all observers
         */
        public void notifyObservers()
        {
            foreach (Observer o in observers)
            {
                o.update();
            }
        }

        /*
         * Notifies all push observers
         */
        public void notifyPushObservers(Object e)
        {
            foreach (Observer o in observers)
            {
                o.pushUpdate(e);
            }
        }

        /*
         * Removes an observer
         */
        public void removeObserver(Observer o)
        {
            observers.Remove(o);
        }
    }
}
